modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusThreadPoolExecutor.java

```diff
@@ -0,0 +1,37 @@
+package im.status.ethereum.module;
+
+import java.util.concurrent.*;
+
+public class StatusThreadPoolExecutor {
+    private static final int NUMBER_OF_CORES =
+            Runtime.getRuntime().availableProcessors();
+    private static final int KEEP_ALIVE_TIME = 1;
+    private static final TimeUnit KEEP_ALIVE_TIME_UNIT = TimeUnit.SECONDS;
+
+    private final BlockingQueue<Runnable> mQueue;
+    private final ThreadPoolExecutor mThreadPool;
+
+    private StatusThreadPoolExecutor() {
+        mQueue = new LinkedBlockingQueue<>();
+
+        mThreadPool = new ThreadPoolExecutor(
+            NUMBER_OF_CORES,
+            NUMBER_OF_CORES,
```

Comment:
what if to increase the second parameter (maximumPoolSize) for more throughput?

Reply:
@audriu Good question. Current setup is case (2) in https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadPoolExecutor.html :

>2. Unbounded queues. Using an unbounded queue (for example a LinkedBlockingQueue without a predefined capacity) will cause new tasks to wait in the queue when all corePoolSize threads are busy. Thus, no more than corePoolSize threads will ever be created. (And the value of the maximumPoolSize therefore doesn't have any effect.)_

Doesn't necessarily maximize throughput, but does minimize CPU context switching and therefore battery use, etc. Not guaranteed that switching to a bounded queue and a larger maximumPoolSize (case 3 in the docs) will increase throughput :

>Using large queues and small pools minimizes CPU usage, OS resources, and context-switching overhead, but can lead to artificially low throughput. If tasks frequently block (for example if they are I/O bound), a system may be able to schedule time for more threads than you otherwise allow. Use of small queues generally requires larger pool sizes, which keeps CPUs busier but may encounter unacceptable scheduling overhead, which also decreases throughput. 

But since most of the tasks under the executor are I/O bound, perhaps it does makes sense to switch to case 3? Would need testing to figure out which performs better?